common/test/acceptance/tests/studio/test_studio_outline.py

```diff
@@ -492,6 +621,167 @@ def test_unchanged_release_date_is_not_saved(self):
 
 
 @attr(shard=3)
+class UnitAccessTest(CourseOutlineTest):
+    """
+    Feature: Units can be restricted and unrestricted to certain groups from the course outline.
+    """
+
+    __test__ = True
+
+    def setUp(self):
+        super(UnitAccessTest, self).setUp()
+        self.group_configurations_page = GroupConfigurationsPage(
+            self.browser,
+            self.course_info['org'],
+            self.course_info['number'],
+            self.course_info['run']
+        )
+        self.content_group_a = "Test Group A"
+        self.content_group_b = "Test Group B"
+
+        self.group_configurations_page.visit()
+        self.group_configurations_page.create_first_content_group()
+        config_a = self.group_configurations_page.content_groups[0]
+        config_a.name = self.content_group_a
+        config_a.save()
+        self.content_group_a_id = config_a.id
+
+        self.group_configurations_page.add_content_group()
+        config_b = self.group_configurations_page.content_groups[1]
+        config_b.name = self.content_group_b
+        config_b.save()
+        self.content_group_b_id = config_b.id
+
+        add_enrollment_course_modes(self.browser, self.course_id, ["audit", "verified"])
+
+    def populate_course_fixture(self, course_fixture):
+        """
+        Create a course with one section, one subsection, and two units
+        """
+        # with collapsed outline
+        self.chap_1_handle = 0
+        self.chap_1_seq_1_handle = 1
+
+        # with first sequential expanded
+        self.seq_1_vert_1_handle = 2
+        self.seq_1_vert_2_handle = 3
+        self.chap_1_seq_2_handle = 4
+
+        course_fixture.add_children(
+            XBlockFixtureDesc('chapter', "1").add_children(
+                XBlockFixtureDesc('sequential', '1.1').add_children(
+                    XBlockFixtureDesc('vertical', '1.1.1'),
+                    XBlockFixtureDesc('vertical', '1.1.2')
+                )
+            )
+        )
+
+    def _set_restriction_on_unrestricted_unit(self, unit):
+        """
+        Restrict unit access to a certain group and confirm that a
+        warning is displayed.  Then, remove the access restriction
+        and verify that the warning no longer appears.
+        """
+        self.assertFalse(unit.has_restricted_warning)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
+        self.assertTrue(unit.has_restricted_warning)
+        unit.set_unit_access('Content Groups', [self.content_group_a_id])
```

Comment:
"set_unit_access" is a confusing method name here because I would assume it always sets the access according to the input arguments. But apparently it is toggling?